### PR TITLE
[luci] Quantization algorithm for InstanceNorm

### PR DESCRIPTION
**Pull request by jinevening**
This adds quantization algorithm for InstanceNorm Op

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #5121
Draft PR: #5197

### CODE REVIEW

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp [hunk 1]
```diff
@@ -86,6 +86,99 @@ void quant_const_values(luci::CircleConst *const_node, float scaling_factor, flo
...
+//   Negative f: f = (-f) * (q - 0) [q = -1, scale = -f, zp = 0]
+void quant_const_per_channel(CircleConst *node, loco::DataType quant_type)
+{
+  assert(node->dtype() == loco::DataType::FLOAT32);
+  assert(node->rank() > 0);
+
+  for (uint32_t i = 0; i < node->rank() - 1; i++)
+  {
+    if (node->dim(i).value() != 1)
+      throw std::runtime_error("Non-channel demension of const node must be 1");
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:112]: (opional) maybe add a comment why there is throw instead of just return and ignore

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp [hunk 2]
```diff
@@ -612,10 +705,31 @@ struct QuantizeWeights final : public luci::CircleNodeMutableVisitor<bool>
...
+  if (granularity == QuantizationGranularity::LayerWise)
+  {
+    quant_const(gamma, output_type);
+    quant_const(beta, output_type);
+  }
+  else
+  {
+    quant_const_per_channel(gamma, output_type);
+    quant_const_per_channel(beta, output_type);
+  }
```
Comment (mhs4670go) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:725]:
```suggestion
  else if (granularity == QuantizationGranularity::ChannelWise)
  {
    quant_const_per_channel(gamma, output_type);
    quant_const_per_channel(beta, output_type);
  }
  else
    throw std::runtime_error("Quantization granularity must be either 'layer' or 'channel'");
```

Comment (seanshpark) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:725]: if `QuantizationGranularity` may provide another type then this change would be good

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp (hunk 1 above)
Comment (jinevening) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp]: I've added the comment.

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp (hunk 2 above)
Comment (jinevening) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp]: I've updated the code.  Thanks.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by jinevening** [comment]
@seanshpark Can you merge this PR? :)

**Pull request merged**